libpius/path.py

```diff
@@ -0,0 +1,20 @@
+'''A set of path functions and variables for the PIUS suite.'''
+
+# vim:shiftwidth=2:tabstop=2:expandtab:textwidth=80:softtabstop=2:ai:
+import os
+
+def which(pgm):
+  path=os.getenv('PATH')
+  for p in path.split(os.path.pathsep):
```

Comment:
We really don't want to do this. gpg is security sensitive and people put crazy shit like `.` in their path. I'm fine have a set of paths we walk through but, not $PATH.

Also, please be mindful of stype - spaces around `=`, etc.

Reply:
This is basically the same thing as what python 3 shutil.which function does, but this is so it will work with python 2 and 3. All you do is specify the program for it to find and it will return the path if there is one, else it will return a blank (None or ""). PATH doesn't like to have relative values only absolute as with any OS (Windows, Mac, and Linux). However if there is something better I can always work on getting the hardcode values for each. This function will work with finding any program as long as the path has access for the directory chain (/usr/local/bin, or /bin, etc.).

Reply:
PATH can *totally* have relative values in it. It's evil to do it, but it's totally supported. Even when it doesn't, people put /tmp and $HOME and other crap in their path:

```
[phild@fuel ~]$ PATH=.:$PATH
[phild@fuel ~]$ cd /tmp
[phild@fuel tmp]$ echo "echo ls from tmp" > ls
[phild@fuel tmp]$ mkdir foo
[phild@fuel tmp]$ echo "echo ls from foo" > foo/ls
[phild@fuel tmp]$ chmod +x ls foo/ls
[phild@fuel tmp]$ ls
ls from tmp
[phild@fuel tmp]$ cd foo
[phild@fuel foo]$ ls
ls from foo
[phild@fuel foo]$ 
```

So maybe like:
```
if windows
  safe_paths += ['C:\Program Files\GnuPG\gpg2.exe']
else
  safe_paths = ['/bin', '/usr/bin']
  if mac
    safe_paths = ['/usr/local/bin']
  end
end
```

Or some such? (conditionals are, obviously, pseudocode).

Reply:
can always check the platform.system to get if it is linux, darwin, or windows. Then use the safe paths that are not complete (windows for instance will be C:\Program Files\) and then ran through the which() function. I will test it out really quick.